### ISSUE TITLE
Ring Alarm support would be a great addition

### ISSUE THREAD

**Issue by jooprooding** [denniedegroot/com.ring#66]
Ring alarm, including all the sensor triggers could add much value to the homey home automation for me.

**Comment by Razfaz72** [comment]
I would be happy too. I would also really appreciate the support from the door and window contacts.

**Comment by Yvarladze** [comment]
Hi Dennie,
I was curious if we can expect the possibility that the ring status can be changed to away and home via homey. I am looking for a solution to automatically switch the cameras on and off because my parents do not want me to be able to watch in the living room, but only when they need help.

**Comment by Dav1dNL** [comment]
@denniedegroot and @kaohlive

I would also like to have the mode change option built in.

As a workaround I now use the ring plugin on homebridge, linked to my Homey. But it would be nice if it could be done directly with Homey.

I am not advanced in programming Homey apps, but the code in the link below may help you build this feature:

https://github.com/dgreif/ring/blob/master/homebridge/location-mode-switch.ts

**Comment by oksdol** [comment]
I would also highly appreciate support for Ring Alarm. It seems to have the best looking alarm keypad. Any thoughts on if we can expect support for it in this app?

**Comment by biginski** [comment]
Also the question to add support for RingAlarm.
In the latest version there is also support for floodlight and sirene, but it is not possible to make a flow which reads the status of the sirene. That would help to make a workaround for alarm situaties of RingAlarm and putting all the lights on (for example).
Thanks for your great work. Regards.

**Comment by kaohlive** [comment]
> but it is not possible to make a flow which reads the status of the sirene. 

I will check out if a event for an active siren is something we can support.

For the Ring Alarm hardware, I see plenty of examples and it does not look to hard to support. But I do not own the Ring Alarm part, only the cams/doorbell part. So its impossible to do any testing. That makes it very repetitive to build it blind, ask for test results and fix my bad coding till it starts to work. That's why I wont implement it sorry.

I am going to work on the location mode support.

**Comment by Airwolf33** [comment]
The branch from @tokreutz is working well with Ring Alarm.
I use it everyday to change modes of my Ring Alarm (Armed / Disarmed / At Home)
If it can help…!

**Comment by kaohlive** [comment]
he should ask dennie to merge then

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by biginski** [comment]
I understand.For me personal it would be great to read the status of the cam alarm.I programmed the Ring alarm so it will also activate the alarm of the camera's.If i can read the status of that alarm i can use it to put all lights on for example.Regards, Ebo.Verzonden vanaf mijn Galaxy
-------- Oorspronkelijk bericht --------Van: kaohlive ***@***.***> Datum: 22-06-2021  21:49  (GMT+01:00) Aan: "denniedegroot/com.ring" ***@***.***> Cc: biginski ***@***.***>, Comment ***@***.***> Onderwerp: Re: [denniedegroot/com.ring] Ring Alarm support would be a great
  addition (#66) 

but it is not possible to make a flow which reads the status of the sirene.

I will check out if a event for an active siren is something we can support.
For the Ring Alarm hardware, I see plenty of examples and it does not look to hard to support. But I do not own the Ring Alarm part, only the cams/doorbell part. So its impossible to do any testing. That makes it very repetitive to build it blind, ask for test results and fix my bad coding till it starts to work. That's why I wont implement it sorry.
I am going to work on the location mode support.

**Comment by kaohlive** [comment]
In my fork you can now find a new version with Location mode support.
It is based on the action cards, on app level you can find a trigger, condition and action card to interact with the modes.
They all provide the Ring locations you have to select from, so it should cover all.

Also added support for a alarm status for a cam that is linked to the siren status. Also removed the link between the tile and the cam so you do not accidentally trigger an siren from the UI.

its in pull request for Dennie, if you can not wait, grab the fork

**Comment by biginski** [comment]
Hi!

Sorry, but I don’t understand this.
I don’t know what a “fork” is. I’m not that familiar with it. Know how to use the Homey and to make flows …..
Can you help me more specific or do I have to wait for an update to make a flow who reads the alarm status of the cam?

Thanks and regards,
Ebo

**Comment by kaohlive** [comment]
Than you have to wait till dennie releases it to homey directly.

**Comment by denniedegroot** [comment]
You can now install the test version: https://homey.app/a/com.amazon.ring/test/

**Comment by kaohlive** [comment]
Now thats quick! Thanks dennie!

**Comment by denniedegroot** [comment]
@kaohlive and thank you for your work!

**Comment by biginski** [comment]
Great. Thanks!

**Comment by tokreutz** [comment]
Sorry for not being very active lately, but this location mode support is not the same as the Ring Alarm support. Supporting the Ring Alarm is more complicated, and I have a suggestion below on how to do that.

I was hoping to spend more time on this app, but I have had other matters to attend to. I suggest that we evolve the app using dgrief/ring api repo, as that has all the functionality we ever need for using ring apis, including the Ring Alarm base station. I have contributed som missing features there as well in order to support existing functionality of the Homey app too. @kaohlive :)

**Comment by kaohlive** [comment]
i checked your fork, very interesting to implement that api indeed. but for me this location mode was a couple of hours work and kept the lightweight.
i also do not have the time to rebuild it based in that api.
the quick test i did, i noticed it bloated the app from small to huge. not sure if thats caused by that api or something else.

**Comment by tokreutz** [comment]
Yes, just supporting location mode took me a couple of hours as well :)
I tried integrating with base station since one of the app users here said it wasnt working for him, and I ended up using the ring api in order to do so.

The app grew to about 15MB due to the ringapi lib, and that was the only downside I could see.
